### DISCUSSION TITLE
Credentials for docker registry not working for podman desktop

### DISCUSSION THREAD

**Discussion by gabrielpantiru** [podman-desktop/podman-desktop#5199] [Q&A]
Hello, 

What credentials needs to be configured on the "Manage registries" section in podman ?

When I try to pull an image I get an error saying that access to image "redis" is denied.
I tried to log in using my account from docker.io but still the same error ( see image)
![image](https://github.com/containers/podman/assets/117661610/29ddb2b9-f394-48ed-a96f-43f989954fa5)

**Comment by rhatdan** [comment]
There should be no configuration needed.

**Comment by gabrielpantiru** [comment]
@rhatdan  Is there any way of seeing more detailed logs of what the underlying issue is?

**Reply by jeffmaury** [reply]
Could you try from the CLI (with trace on): podman pull redis --log-level trace

**Reply by gabrielpantiru** [reply]
@jeffmaury Sure here is the result of the command:
`podman pull redis --log-level trace
time="2023-12-14T18:28:00+02:00" level=info msg="C:\\Program Files\\RedHat\\Podman\\podman.exe filtering at log level trace"
time="2023-12-14T18:28:00+02:00" level=debug msg="Called pull.PersistentPreRunE(C:\\Program Files\\RedHat\\Podman\\podman.exe pull redis --log-level trace)"
time="2023-12-14T18:28:00+02:00" level=debug msg="SSH Ident Key \"C:\\\\Users\\\\pantiru.angelo\\\\.ssh\\\\podman-machine-default\" SHA256:oeBGTX1SQAohjdsadgreKGsjguLtBCdh03kqQfxrhEc ssh-ed21996"
time="2023-12-14T18:28:00+02:00" level=debug msg="DoRequest Method: GET URI: http://d/v4.7.2/libpod/_ping"
time="2023-12-14T18:28:00+02:00" level=debug msg="Loading registries configuration \"/etc/containers/registries.conf\""
time="2023-12-14T18:28:00+02:00" level=debug msg="Found credentials for docker.io in credential helper containers-auth.json in file C:\\Users\\pantiru.angelo\\.config\\containers\\auth.json"
time="2023-12-14T18:28:00+02:00" level=debug msg="No credentials matching registry.ppearson.io found in C:\\Users\\pantiru.angelo\\.config\\containers\\auth.json"
time="2023-12-14T18:28:00+02:00" level=debug msg="No credentials matching registry.ppearson.io found in C:\\Users\\pantiru.angelo\\.config\\containers\\auth.json"
time="2023-12-14T18:28:00+02:00" level=debug msg="Found an empty credential entry \"registry.ppearson.io\" in \"C:\\\\Users\\\\pantiru.angelo\\\\.docker\\\\config.json\" (an unhandled credential helper marker?), moving on"
time="2023-12-14T18:28:00+02:00" level=debug msg="No credentials matching registry.ppearson.io found in C:\\Users\\pantiru.angelo\\.dockercfg"
time="2023-12-14T18:28:00+02:00" level=debug msg="No credentials for registry.ppearson.io found"
time="2023-12-14T18:28:00+02:00" level=debug msg="DoRequest Method: POST URI: http://d/v4.7.2/libpod/images/pull"
Resolving "redis" using unqualified-search registries (/etc/containers/registries.conf.d/999-podman-machine.conf)
Trying to pull docker.io/library/redis:latest...
Error: initializing source docker://redis:latest: pinging container registry registry-1.docker.io: Get "https://registry-1.docker.io/v2/": dial tcp: lookup registry-1.docker.io: Temporary failure in name resolution
time="2023-12-14T18:28:40+02:00" level=debug msg="Shutting down engines"`

It seems the podman machine doesnt have the dns of the "registry-1.docker.io" host.  If i try to ping it from my windows powershel I get: 
`ping registry-1.docker.io

Pinging registry-1.docker.io [34.226.69.105] with 32 bytes of data:
Request timed out.
Request timed out.
Request timed out.
Request timed out.`
I also tried opening the link on the browser "https://registry-1.docker.io/v2/" and I get 401, which probably means only the podmachine has some restricted access.

Do you have any ideas what I could try in this case? 

Do you think if i try to add 8.8.8.8 in the etc folder of podman distribution it will work, or there can be other restrictions that can cause this? Many thanks. Here is the output of wsl -l :
Windows Subsystem for Linux Distributions:
docker-desktop-data (Default)
docker-desktop
Ubuntu
podman-machine-default

**Reply by jeffmaury** [reply]
Can you run podman machine inspect ? And also what is the content of /etc/resolv.conf in the WSL machine ?

**Reply by gabrielpantiru** [reply]
@jeffmaury the result of the podman machine inspect is:
` podman machine inspect
[
     {
          "ConfigPath": {
               "Path": "C:\\Users\\pantiru.angelo\\.config\\containers\\podman\\machine\\wsl\\podman-machine-default.json"
          },
          "ConnectionInfo": {
               "PodmanSocket": null,
               "PodmanPipe": {
                    "Path": "\\\\.\\pipe\\podman-machine-default"
               }
          },
          "Created": "2023-09-12T16:14:02.5106071+03:00",
          "Image": {
               "IgnitionFilePath": {
                    "Path": ""
               },
               "ImageStream": "35",
               "ImagePath": {
                    "Path": "C:\\Users\\pantiru.angelo\\.local\\share\\containers\\podman\\machine\\wsl\\podman-machine-default_fedora-podman-amd64-v38.0.5.tar"
               }
          },
          "LastUp": "2023-12-14T19:01:05.5260103+02:00",
          "Name": "podman-machine-default",
          "Resources": {
               "CPUs": 8,
               "DiskSize": 2722103296,
               "Memory": 5252689920
          },
          "SSHConfig": {
               "IdentityPath": "C:\\Users\\pantiru.angelo\\.ssh\\podman-machine-default",
               "Port": 64607,
               "RemoteUsername": "user"
          },
          "State": "running",
          "UserModeNetworking": false,
          "Rootful": true
     }
]`

also the result of the files /etc/wsl.conf and /etc/resolv.conf:
`cat wsl.conf
[network]
generateResolvConf = false

cat /etc/resolv.conf
nameserver 8.8.8.8
nameserver 8.8.4.4` 
I remember i modified this a few weeks ago when I first tried podman, to have those ip's and to not regenerate the conf in wsl.conf, but still did not fixed the problem

**Reply by gabrielpantiru** [reply]
Also I paste below the content of the C:\\\\Users\\\\pantiru.angelo\\\\.docker\\\\config.json\ file from my windows i because I saw in the error log of the command "podman pull redis --log-level trace" that i pasted above, there is also mention about this as being a issue "Found an empty credential entry \"registry.ppearson.io\" in \"C:\\\\Users\\\\pantiru.angelo\\\\.docker\\\\config.json\" (an unhandled credential helper marker?),":
`{
	"auths": {
		"registry.ppearson.io": {}
	},
	"credsStore": "desktop",
	"currentContext": "default"
}`

**Comment by jeffmaury** [comment]
Can you try to restart the Podman machine ?

**Reply by gabrielpantiru** [reply]
@jeffmaury  I stoped the podman and started it again from the windows tray and the same result.

A short update, I noticed the credential files it was from docker "C:\\Users\\pantiru.angelo\\.docker\\config.json" and if i delete that config, now i get different logs for the command "pull redis --log-level trace":

`time="2023-12-14T19:30:05+02:00" level=info msg="C:\\Program Files\\RedHat\\Podman\\podman.exe filtering at log level trace"
time="2023-12-14T19:30:05+02:00" level=debug msg="Called pull.PersistentPreRunE(C:\\Program Files\\RedHat\\Podman\\podman.exe pull redis --log-level trace)"
time="2023-12-14T19:30:05+02:00" level=debug msg="SSH Ident Key \"C:\\\\Users\\\\pantiru.angelo\\\\.ssh\\\\podman-machine-default\" SHA256:oeBGTX1SQAohjdsadgreKGsjguLtBCdh03kqQfxrhEc ssh-ed21996"
time="2023-12-14T19:30:05+02:00" level=debug msg="DoRequest Method: GET URI: http://d/v4.7.2/libpod/_ping"
time="2023-12-14T19:30:05+02:00" level=debug msg="Loading registries configuration \"/etc/containers/registries.conf\""
time="2023-12-14T19:30:05+02:00" level=debug msg="Found credentials for docker.io in credential helper containers-auth.json in file C:\\Users\\pantiru.angelo\\.config\\containers\\auth.json"
time="2023-12-14T19:30:05+02:00" level=debug msg="DoRequest Method: POST URI: http://d/v4.7.2/libpod/images/pull"
Resolving "redis" using unqualified-search registries (/etc/containers/registries.conf.d/999-podman-machine.conf)
Trying to pull docker.io/library/redis:latest...
Error: initializing source docker://redis:latest: pinging container registry registry-1.docker.io: Get "https://registry-1.docker.io/v2/": dial tcp: lookup registry-1.docker.io: Temporary failure in name resolution
time="2023-12-14T19:30:45+02:00" level=debug msg="Shutting down engines"`

Let me know if there is any other commands I could try to investigate the issue and thanks for your input so far :)

**Comment by gabrielpantiru** [comment]
@jeffmaury Hello Jeff, can you please provide me the output of your machine for the command "pull redis --log-level trace" ( when you don't have the docker started) and what windows build version do you have? I found a [ticket ](https://github.com/microsoft/WSL/issues/6427)that multiple people have the issue accessing the internet from wsl 2 on windows build 1909, but with wsl1 it's working.

**Reply by jeffmaury** [reply]
Here's my output:

```
podman pull redis --log-level trace
time="2024-01-02T11:34:33+01:00" level=info msg="podman filtering at log level trace"
time="2024-01-02T11:34:33+01:00" level=debug msg="Called pull.PersistentPreRunE(podman pull redis --log-level trace)"
time="2024-01-02T11:34:33+01:00" level=debug msg="SSH Ident Key \"C:\\\\Users\\\\Jeff\\\\.ssh\\\\podman-machine-default\" SHA256:YCOHnwm5U9ZdeOhsUnvqH+LhQI+mEbB86eEDvbiithg ssh-ed25519"
time="2024-01-02T11:34:33+01:00" level=debug msg="DoRequest Method: GET URI: http://d/v4.8.2/libpod/_ping"
time="2024-01-02T11:34:33+01:00" level=debug msg="Loading registries configuration \"/etc/containers/registries.conf\""
time="2024-01-02T11:34:33+01:00" level=debug msg="DoRequest Method: POST URI: http://d/v4.8.2/libpod/images/pull"
Resolving "redis" using unqualified-search registries (/etc/containers/registries.conf.d/999-podman-machine.conf)
Trying to pull docker.io/library/redis:latest...
Getting image source signatures
Copying blob sha256:af107e978371b6cd6339127a05502c5eacd1e6b0e9eb7b2f4aa7b6fc87e2dd81
Copying blob sha256:797591c7970a3db095434a6fd0673ae80c40f814da1c1b39787faf63b8dee51d
Copying blob sha256:e3b2691a4104bbe54bf413a8ae7b8b442d8fc00af86067a4174261f050cab9a5
Copying blob sha256:bf7f0c8796d3701fef70b4099c3302f3a9e017cc615adf5046f95460fd4b921b
Copying blob sha256:190b4d7a237a66227f26efb851e9731674003d6337dc61e7f3a41aabd8eaf479
Copying blob sha256:b031def5f2c4a8515f6b4e609686a0a9846d8a4308bf6c1d9f41b0bb627275f1
Copying blob sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1
Copying blob sha256:45ce3854ac9a2a8cebfac64e81e22a7047de8acc8a255ac6979272bb91eb645a
Copying config sha256:e40e2763392d8dea3270841365bf740c7157fb6076a09b43e669aa4159fc5de3
Writing manifest to image destination
e40e2763392d8dea3270841365bf740c7157fb6076a09b43e669aa4159fc5de3
time="2024-01-02T11:34:46+01:00" level=debug msg="Called pull.PersistentPostRunE(podman pull redis --log-level trace)"
time="2024-01-02T11:34:46+01:00" level=debug msg="Shutting down engines"
```

**Reply by gabrielpantiru** [reply]
And the ping command it's working inside the podman-machine-default wsl ? Eg ping docker.io

**Comment by symplrdudley** [comment]
I think I am in a similar situation where Podman Desktop UI says "Create Container" then I choose from existing image. Then I go to pull an image, and Podman Desktop says that I do not have any registries configured. So I go and look at it, and there are 5 of them which say I need credentials. However, I can pull the same image from the CLI. 

Why do I need credentials in Podman Desktop and not the CLI?

**Reply by benoitf** [reply]
to pull images you don't need any credentials unless the image you want to pull is a private image

**Reply by symplrdudley** [reply]
I know that as I can do that via the CLI,b ut in Docker Desktop, I cannot pull images because "registries are not configured" but when I go to configure them, I see docker.io but when I "configure" it, it only asks for username and password, but no place to put the URL for the image to pull.

So that's why I ask.

**Reply by benoitf** [reply]
the registries page is where you setup the credentials for the registry like index.docker.io or docker.io

then when pulling an image like index.docker.io/myuser/my-private-image (in the images page, top right pull button) it'll use the credentials set in the registries page

**Comment by staylet** [comment]
I have the same problem. I set up a local [Docker registry](https://distribution.github.io/distribution/) with basic auth and a self-signed certificate. Domain resolution is handled via the Hosts file, and the certificate is trusted ([Adding certificates to a Podman machine](https://podman-desktop.io/docs/podman/adding-certificates-to-a-podman-machine)). The `podman pull` command works in the podman machine, and the registry is added successfully in Settings > Registries. However, I cannot pull images in the Podman Desktop UI.

The error always shows:
`Error while pulling image from podman-machine-default: access to image "{my domain}:5443/base/ubuntu:24.04" is denied (401 error). Can also be that the registry requires authentication.`
